solidity/ts/generateHookSalt.ts

```diff
@@ -0,0 +1,14 @@
+import { HOOK_IMPLEMENTATION_CODE } from './testsuite/simulator/utils/constants.js';
+import { getAugurConstantProductMarketRouterAddress } from './testsuite/simulator/utils/utilities.js'
+
+console.log(`Searching for salt`)
+
+let salt = 0n
+while (true) {
+	const addressAttempt = getAugurConstantProductMarketRouterAddress(salt)
+	if (addressAttempt.endsWith(HOOK_IMPLEMENTATION_CODE)) {
+		console.log(`!!! Salt found: ${salt}`)
+		process.exit()
+	}
+	salt++;
```

Comment:
could this be made into a function that is run when deployment is done? Now everytime we update router, we need to run this to get the salt. It could then be cached so its always only searched again if the router has changed

Reply:
It can take quite a bit so it isnt something I'd want personally for each deployment since we do that in every test. Im sure we'll end up coming back to this though in some way to make it less cumbersome.